### ISSUE TITLE
Dose `seed` affect the training when we set customized train/val/test set?

### ISSUE THREAD

**Issue by WeiJunZhou23** [HICAI-ZJU/KANO#12]
When we set up customized train/val/test set and didn't set a value for `seed`, does the `seed` argument still affect how the model is being trained? In my case, I didn't set a value for `seed`, but I can still see `Splitting data with seed 1` in the `train.log`, and model performance in different runs is slightly different.
Here is my command:
`command: python train.py --data_path './dataset_train.csv' --separate_val_path './dataset_val.csv' --separate_test_path './dataset_test.csv' --dataset_type classification --epochs 100 --num_runs 3 --gpu 0 --batch_size 256 --init_lr '1e-4' --ensemble_size 1 --step functional_prompt --exp_name finetune --exp_id activity --checkpoint_path './dumped/pretrained_graph_encoder/original_CMPN_0623_1350_14000th_epoch.pkl' --exp_id "activity"`

**Comment by ZJU-Fangyin** [comment]
Hi, when you specify your own separate train/val/test sets, the model will be trained directly on the data you provided, and the data splitting process is not involved. Therefore, the `seed` argument does not affect the data the model sees during training in this case.

The `Splitting data with seed 1` log is a default message that appears in the log, but it doesn't apply if you provide your own separate datasets. The slight variation in model performance across different runs could be due to other factors, such as stochasticity in the optimization process, initialization of the model parameters, etc.

**Comment by WeiJunZhou23** [comment]
Thank you for your clarification, I really appreciate your work.
